### ISSUE TITLE
Alternatives to Firebase, PoC, ideas

### ISSUE THREAD

**Issue by vimark1** [vimark1/typist#102]
I like Firebase for this project, because it allows us get away with not using a server, yet, at the same time, this project is becoming more and more dependent on it.  I am keen to find any alternatives that we can look into.

I am interested in implementing a PoC on using the blockchain technology. Some more details on the article here:
https://medium.com/@takleakshar/how-to-build-a-decentralized-full-stack-app-in-ethereum-and-react-42e63d45a208

**Comment by clayreimann** [comment]
Do you want to get away from Firebase because you don't like Firebase or because the Firebase code is becoming ever more intertwined with our application code? I think the code base needs a refactor (and have opened #104) to track that.

After reading your article I don't see how building a blockchain-based API will help. In the article there is a server process running that the UI talks to. I did some looking and didn't find much in the way of free API hosting that could replace Firebase. If you're willing to build an API to back the app we could use
* [heroku](https://www.heroku.com/nodejs)
* [Zeit now](https://zeit.co/docs)

**Comment by ccampean** [comment]
I am in the same situation of using Firebase but when I am ready to switch I will take this approach into consideration https://medium.com/one-tap-software/why-im-leaving-firebase-for-pouchdb-or-am-i-73a12390bfef + https://forum.ionicframework.com/t/what-are-the-differences-between-couchdb-firebase-parse-and-pouchdb/36001/5

**Comment by vimark1** [comment]
@clayreimann 

> After reading your article I don't see how building a blockchain-based API will help. In the article there is a server process running that the UI talks to

yes, maybe we can run on zeit, not heroku though.  If we kept the server very small I think it wouldn't be too bad.  Also, there may be some blockchain as a service type of thing.  I think it would be interesting to see what can be done with this technology though.

**Comment by xuman2019** [comment]
[DB3 Network](https://github.com/dbpunk-labs/db3) - Decentralized Firebase Firestore Alternative.
